BootloaderCorePkg/Include/Library/PciEnumerationLib.h

```diff
@@ -8,6 +8,8 @@
 #ifndef __PCI_ENUMERTION_LIB_H__
 #define __PCI_ENUMERTION_LIB_H__
 
+typedef VOID   (*PLATFORM_PCI_ENUM_HOOK) (UINT32 SecBus);

```

Comment:
Add EFIAPI
`(EFIAPI *PLATFORM_PCI_ENUM_HOOK)`

Parameters should be `Bus, Device, Func, Phase`.

```
typedef VOID   (*PLATFORM_PCI_ENUM_HOOK) (UINT32 SecBus);
->
typedef VOID   (EFIAPI *PLATFORM_PCI_ENUM_HOOK) (UINT8 Bus, UINT8 Device, UINT8 Func, EFI_PCI_CONTROLLER_RESOURCE_ALLOCATION_PHASE Phase);
```

Reply:
Do we have a better place in BootloaderCommonPkg instead of BootloaderCorePkg?

Reply:
Theres PlatformHookLib.h in CommonPkg. How does it sound?

Reply:
PlatformHookLib is a lib class defined by EDK2.  So let us leave it.
Since this PCD is defined in BootloaderCorePkg.dec,  the related definition should also be in the same package.  PciEnumerationLib.h looks good to me.  Any special reason why we want to move it into BootlaoderCommonPkg ?  Which library in BootlaoderCommonPkg will use this ?

Reply:
This was to avoid BootloaderCorePkg reference from Platform package directly. We have PciRootBridgeInfoGuid.h in BootloaderCommonPkg for some PciEnumeration interfaces, but it looks not suitable file. I am still thinking we need to avoid CorePkg reference from Platform/Silicon package. Any advices on this?

Reply:
When we defined the package dependency rules,  at that time, we allowed Platform/Silicon library to consume Core package.  And there are many instances already.  Do you see any issue to allow that ?

Reply:
Might be my bad memory. I thought the dependency rule does not allow Platform/Silicon to consume Core package like these.
- Common: independent to others
- Core: Common
- Platform: Common and Silicon
- Silicon: Common

If Core package can be referenced from Platform/Silicon, ACPI, MP or etc. libraries can be also used. What about limiting BootloaderCoreLib only?

Reply:
Let's take this dependency rule discussion to offline. Let me merge this PR. Thanks.